aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -243,7 +243,7 @@ def _save_to_cache(self, data_record: DataRecord):
     def _retrieve_from_cache(self, idempotency_key: str):
         cached_record = self._cache.get(idempotency_key)
         if cached_record:
-            if not cached_record.is_expired:
+            if all((not cached_record.is_expired, not cached_record.status == "INPROGRESS")):
```

Comment:
It might be more correct to use the `STATUS_CONSTANTS`

```suggestion
            if all((not cached_record.is_expired, not cached_record.status == STATUS_CONSTANTS["INPROGRESS"])):
```

OR remove the check all together, as we are already preventing this from happening in `get_record` 

```suggestion
            if not cached_record.is_expired:
```